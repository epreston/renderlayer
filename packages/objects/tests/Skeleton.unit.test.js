import { describe, expect, it, test, vi } from 'vitest';

import { Skeleton } from '../src/Skeleton.js';

describe('Objects', () => {
  describe('Skeleton', () => {
    test('constructor', () => {
      const object = new Skeleton();
      expect(object).toBeDefined();
    });

    test('uuid', () => {
      const object = new Skeleton();
      expect(object.uuid).to.be.a('string');
      expect(object.uuid).to.have.length(36);
    });

    test.todo('bones', () => {
      // implement
    });

    test.todo('boneInverses', () => {
      // implement
    });

    test.todo('boneMatrices', () => {
      // implement
    });

    test.todo('boneTexture', () => {
      // implement
    });

    test.todo('boneTextureSize', () => {
      // implement
    });

    test.todo('frame', () => {
      // implement
    });

    test.todo('init', () => {
      // implement
    });

    test.todo('calculateInverses', () => {
      // implement
    });

    test.todo('pose', () => {
      // implement
    });

    test.todo('update', () => {
      // implement
    });

    test.todo('clone', () => {
      // implement
    });

    test.todo('computeBoneTexture', () => {
      // implement
    });

    test.todo('getBoneByName', () => {
      // implement
    });

    test('dispose', () => {
      const object = new Skeleton();
      object.dispose();

      expect(object).toBeDefined();
    });

    test.todo('fromJSON', () => {
      // implement
    });

    test.todo('toJSON', () => {
      // implement
    });
  });
});
