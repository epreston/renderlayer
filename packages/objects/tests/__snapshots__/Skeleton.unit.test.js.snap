// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Objects > Skeleton > toJSON 1`] = `
{
  "boneInverses": [
    [
      1,
      0,
      0,
      0,
      0,
      1,
      0,
      0,
      0,
      0,
      1,
      0,
      0,
      0,
      0,
      1,
    ],
    [
      1,
      0,
      0,
      0,
      0,
      1,
      0,
      0,
      0,
      0,
      1,
      0,
      0,
      0,
      0,
      1,
    ],
    [
      1,
      0,
      0,
      0,
      0,
      1,
      0,
      0,
      0,
      0,
      1,
      0,
      0,
      0,
      0,
      1,
    ],
  ],
  "bones": [
    "67f50833-ee0f-4d7a-b326-751057ac4d41",
    "53c519fb-0ede-4ebd-9861-ebca457e4654",
    "62bbd12f-0a27-42fe-a2a7-f040d91b2152",
  ],
  "metadata": {
    "generator": "Skeleton.toJSON",
    "type": "Skeleton",
    "version": 4.5,
  },
  "uuid": "14e0bc18-e2f2-4d3b-ab56-90561fcf1afd",
}
`;
